### PR TITLE
Shifting categories of discrete numerical columns to start at 0

### PR DESCRIPTION
**Pull request by Ta7ar**
Fixed error where marginal distribution detection was failing if a discrete numerical column had non 0 min value.

### CODE REVIEW

File: synthetic_data/marginal_dist.py [hunk 1]
```diff
@@ -70,13 +70,16 @@ def _detect_dist_discrete(col_stats):
     observed_freq = [categorical_count[category] for category in categories]
     sample_size = col_stats["sample_size"]
 
+    # Shift categories so they start at 0
+    categories = list(range(len(categories)))
```
Comment (taylorfturner) [inline synthetic_data/marginal_dist.py:74]: why is this needed? Overwriting this variable with the same variable name is a bit of an alarm for me

**Review by taylorfturner — COMMENTED** [review]
one comment

**Comment by taylorfturner** [comment]
will approve once test is added to ensure success on edge case @Ta7ar

**Comment by Ta7ar** [comment]
Closing this PR for the time being, will open with the same changes after I made the changes to a branch other than `Ta7ar:main`
